eng/common/pipelines/templates/steps/prepare-pipelines.yml

```diff
@@ -2,6 +2,9 @@ parameters:
   - name: Repository
     type: string
     default: $(Build.Repository.Name)
+  - name: Branch
```

Comment:
Why not use our default branch template and just query for the default branch before running the tool?

Reply:
Swapped over to use it. I also removed the checkout in ```pipeline-generation``` because if we just clone we should pick up the default branch.